src/domain/inventory/locations/new/index.tsx

```diff
@@ -0,0 +1,192 @@
+import {
+  AdminPostStockLocationsReq,
+  SalesChannel,
+  StockLocationAddressDTO,
+  StockLocationAddressInput,
+} from "@medusajs/medusa"
+import {
+  useAdminAddLocationToSalesChannel,
+  useAdminCreateStockLocation,
+} from "medusa-react"
+import { useForm } from "react-hook-form"
+import Button from "../../../../components/fundamentals/button"
+import CrossIcon from "../../../../components/fundamentals/icons/cross-icon"
+import FocusModal from "../../../../components/molecules/modal/focus-modal"
+import Accordion from "../../../../components/organisms/accordion"
+import { useFeatureFlag } from "../../../../context/feature-flag"
+import useNotification from "../../../../hooks/use-notification"
+import { getErrorMessage } from "../../../../utils/error-messages"
+import { nestedForm } from "../../../../utils/nested-form"
+import AddressForm from "../components/address-form"
+import GeneralForm, { GeneralFormType } from "../components/general-form"
+import SalesChannelsForm from "../components/sales-channels-form"
+
+type NewLocationForm = {
+  general: GeneralFormType
+  address: StockLocationAddressDTO
+  salesChannels: {
+    channels: SalesChannel[]
+  }
+}
+
+const NewLocation = ({
+  onClose,
+  refetch,
+}: {
+  onClose: () => void
+  refetch: () => void
+}) => {
+  const form = useForm<NewLocationForm>({
+    defaultValues: {
+      general: {
+        name: "",
+      },
+      address: undefined,
+      salesChannels: {
+        channels: [],
+      },
+    },
+    reValidateMode: "onBlur",
+    mode: "onBlur",
+  })
+  const { handleSubmit, formState } = form
+
+  const notification = useNotification()
+  const { isFeatureEnabled } = useFeatureFlag()
+
+  const { mutate } = useAdminCreateStockLocation()
```

Comment:
question/suggestion: can we use `mutateAsync` here as well, so we use `async/await` throughout `handleSubmit` which would be a bit cleaner and more readable